bpf/include/bpf/helpers_xdp.h

```diff
@@ -38,6 +38,10 @@ static int BPF_STUB(xdp_store_bytes, struct xdp_md *xdp, __u32 off,
 		    const void *from, __u32 len, __u32 flags);
 #endif
 
+#ifdef HAVE_XDP_GET_BUFF_LEN
```

Comment:
Is this `#ifdef` strictly necessary? The ones above conditionally emit stubs. As long as all the uses are gated, we shouldn't need this.

Reply:
Right, I suppose we don't need this `#ifdef` (as no-one should be calling the function unless `HAVE_XDP_GET_BUFF_LEN` is set).

On the other hand it's nicely self-documenting, and should help to catch users that **don't** have the check? So all things being equal, I think we benefit from having it.

Reply:
Up to you. I'm just thinking of a future where we would rely more on runtime adjustments to BPF code instead of recompiling the world for every minor config change, but this is looking less achievable by the day. This would be one less `#ifdef` we need to remove.

Reply:
Thanks for taking it into consideration! :+1: